xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/PVR.h

```diff
@@ -2420,6 +2420,16 @@ class ATTR_DLL_LOCAL CInstancePVRClient : public IAddonInstance
   }
   //----------------------------------------------------------------------------
 
+  //============================================================================
+  /// @brief The currently playing stream has been closed
+  ///
+  /// @remarks Called if both @ref PVRCapabilities::SetHandlesInputStream() or
+  /// @ref PVRCapabilities::SetHandlesDemuxing() are set to false. Allows add-ons
+  /// to do any cleanup required prior to a stream being opened.
+  ///
+  virtual void StreamClosed() {}
```

Comment:
Should return a `PVR_ERROR` and the default implementation should return `PVR_ERROR_NOT_IMPLEMENTED`.